lib/iex/lib/iex/h_elixir.ex

```diff
@@ -0,0 +1,143 @@
+defmodule IEx.H_Elixir do
+  @moduledoc """
+  This module duplicates the prior behaviour of the iex h
+  command using the documentation interface.
+
+  This interface consists of the functions
+
+  * `documentation(module)`
+
+  * `documentation(module, function)`
+
+  * `documentation(module, function, arity)`
+
+  All of these functions return a tuple of status and doc_list.
+
+  * `{:found, doc_list}`     - Documentation was found
+
+  * `{:not_found, doc_list}` - Documentation was not found and
+                              this helper is sure it can't be found
+
+  * `{:unknown, doc_list}`   - Helper does not know how to find docs for
+                               arguments given.
+
+  doc_list consists of a list of `{header, markdown}` tuples for each
+  item found.
+  """
+
+  def documentation(module) do
+    case is_elixir?(module) do
+      true -> get_doc(module)
+      _   -> {:unknown, [{inspect(module), ""}]}
+    end
+  end
+
+  def documentation(module, function) do
+    case is_elixir?(module) do
+      true -> get_doc(module, function)
+      _   -> {:unknown, [{inspect(module), ""}]}
+    end
+  end
+
+  def documentation(module, function, arity) do
+    case is_elixir?(module) do
+      true -> get_doc(module, function, arity)
+      _  -> {:unknown, [{inspect(module), ""}]}
+    end
+  end
+
+
+  def is_elixir?(module) do
+    module
+    |> Atom.to_string
+    |>String.starts_with?("Elixir.")
+  end
+
+  def get_doc(module) when is_atom(module) do
+    docs = Code.get_docs(module, :moduledoc)
+    case docs do
+      nil -> {:not_found, [{inspect(module), "No moduledocs found\n"}]}
+      _   -> {:found, [{inspect(module), elem(docs, 1)}]}
+    end
+  end
+
+  def get_doc(module, function) when is_atom(module) and is_atom(function) do
+    docs = Code.get_docs(module, :docs)
+    case docs do
+      nil -> {:not_found, [{"#{inspect module}.#{function}", "No documentation for #{inspect module}.#{function} found\n"}]}
```

Comment:
regarding all the :no_found messages, wouldn't it be more covenient to just add `\n` when printing it that add it every time you define the message? I mean, there are more changes that you will defining one of them, and missing it when you print it, and if the latter happens it will be easier to spot and fix.